### PR TITLE
Arizona Long-term Capital Gains Subtraction

### LINKED ISSUE #2993: Arizona Long-term Capital Gains Subtraction
https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01022.htm

### PR DESCRIPTION
**Pull request by JudyZ13**
Fixes #2993

### Summary
:sparkles::page_facing_up::white_check_mark:

This pull request adds a new feature to the policy engine: the Arizona net capital gains subtraction. It introduces a new parameter, variable, and test cases for this feature, and updates the changelog accordingly.

> _New feature added_
> _`az_net_capital_gains_subtractions`_
> _Springtime for taxpayers_

### Walkthrough
*  Add a new feature to allow subtracting a portion of net capital gains from Arizona income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4), [link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-e212534be1e7a905a00bb945a1f22a89c960a028ec485990bb6bbfb9ca5c9101R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-5355bc4a4efdf5f25da08f52446cf9b2c889afe3898659b74a33101bb8d474d7R1-R24), [link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-747a73a39a784f83909cc11d2911b634f5f5772f3cc1cb4d087f59c1232c79d8R1-R17))
  - Create a new parameter file `subtraction_rate.yaml` to define the rate of subtraction for different years ([link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-e212534be1e7a905a00bb945a1f22a89c960a028ec485990bb6bbfb9ca5c9101R1-R15))
  - Create a new variable file `az_net_capital_gains_subtractions.py` to implement the formula for calculating the subtractions based on the parameter and the tax unit's capital gains ([link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-5355bc4a4efdf5f25da08f52446cf9b2c889afe3898659b74a33101bb8d474d7R1-R24))
  - Create two test cases in `az_net_capital_gains_subtractions.yaml` to verify the expected output of the variable for different scenarios ([link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-747a73a39a784f83909cc11d2911b634f5f5772f3cc1cb4d087f59c1232c79d8R1-R17))
  - Update the changelog entry to indicate the new feature and the minor version bump ([link](https://github.com/PolicyEngine/policyengine-us/pull/2994/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  reference:
+  - title: 43-1022. Subtractions from Arizona gross income (22)
+    href: https://www.azleg.gov/ars/43/01022.htm
+  - title: 2022 Arizona Resident Personal Income Tax Return Form Line24
+    href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140.pdf#page=1
+values:
+  2013-01-01: 0.1
+  2014-01-01: 0.2
+  2015-01-01: 0.25
+  2022-01-01: 0.25
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml:15]:
```suggestion
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+description: Arizona provides filers a subtraction on long-term capital gains by this rate.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml:1]:
```suggestion
description: Arizona multiplies the long term capital gains by the following rate under the net capital gains deduction.
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  label: Arizona subtraction rate of long-term net capital gains
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml:3]:
```suggestion
  label: Arizona long-term net capital gains subtraction rate
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml:1]: rename to rate.yaml

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+description: Arizona provides filers a subtraction on long-term capital gains by this rate.
+metadata:
+  label: Arizona subtraction rate of long-term net capital gains 
+  period: year
+  unit: /1
+  reference:
+  - title: 43-1022. Subtractions from Arizona gross income (22)
+    href: https://www.azleg.gov/ars/43/01022.htm
+  - title: 2022 Arizona Resident Personal Income Tax Return Form Line24
+    href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/subtraction_rate.yaml:10]: add the tax form instructions as a reference as well

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtractions.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class az_net_capital_gains_subtractions(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtractions.py:4]:
```suggestion
class az_net_capital_gains_subtraction(Variable):
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtractions.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona net capital gains subtractions"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtractions.py:7]:
```suggestion
    label = "Arizona net capital gains subtraction"
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.net_capital_gains
+        person = tax_unit.members
+
+        net_cg_subtraction = tax_unit("net_capital_gain", period)
+
+        long_term_capital_gains = person("long_term_capital_gains", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:21]:
```suggestion
        long_term_capital_gains = add(tax_unit, period, ["long_term_capital_gains"])
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class az_net_capital_gains_subtraction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona net capital gains subtraction"
+    unit = USD
+    documentation = "https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01022.htm"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:9]:
```suggestion
    documentation = "https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i.pdf#page=31"
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.net_capital_gains
+        person = tax_unit.members
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:17]:
```suggestion
```

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+- name: net capital gains is $1000 and long term capital gains is $20,000
+  period: 2023
+  input:
+    state_code: AZ
+    net_capital_gain: 1_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml:5]:
```suggestion
```

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+- name: long term capital gains is $20,000
+  period: 2023
+  input:
+    state_code: AZ
+    long_term_capital_gains: 20_000
+  output:
+    az_net_capital_gains_subtraction: 5000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml:7]:
```suggestion
    az_net_capital_gains_subtraction: 5_000
```

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+- name: long term capital gains is $20,000
+  period: 2023
+  input:
+    state_code: AZ
+    long_term_capital_gains: 20_000
+  output:
+    az_net_capital_gains_subtraction: 5_000
+
+- name: long term capital gains is zero
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml:9]:
```suggestion
- name: long term capital gains are zero
```

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+- name: long term capital gains is $20,000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml:1]:
```suggestion
- name: Long term capital gains are $20,000
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.net_capital_gains
+
+        long_term_capital_gains = add(
+            tax_unit, period, ["long_term_capital_gains"]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:19]: long-term capital gains are multiplied by the rate but are capital gains in general included in the subtraction? Please double check form and law and add a comment in the .py code

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 15]
```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class az_net_capital_gains_subtraction(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:4]:
```suggestion
class az_long_term_capital_gains_subtraction(Variable):
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona net capital gains subtraction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:7]:
```suggestion
    label = "Arizona long-term capital gains subtraction"
```

File: changelog_entry.yaml [hunk 17]
```diff
@@ -1,4 +1,4 @@
 - bump: minor
   changes:
     added:
-    - Kentucky family size tax credit.
+    - Arizona net capital gains subtraction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Arizona long-term capital gains subtraction.
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.py:1]: rename file az_long_term_capital_gains_subtraction.py

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/deductions/net_capital_gains/az_net_capital_gains_subtraction.yaml:1]: rename file to az_long_term_capital_gains_subtraction.yaml

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.net_capital_gains
+
+        long_term_capital_gains = add(
+            tax_unit, period, ["long_term_capital_gains"]
+        )
+        # net capital gain derived from investment in a qualified small business should also be subtracted
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py:21]:
```suggestion
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py:1]: can make you create a folder structure like this:

az/tax/income/subtractions/capital_gains/az_long_term_capital_gains_subtraction.py

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py:1]: same for tests

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/rate.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/rate.yaml:1]: move to subtractions folder (if that's what it is)

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/rate.yaml [hunk 19]
```diff
@@ -0,0 +1,16 @@
+description: Arizona multiplies the long term capital gains by the following rate under the net capital gains deduction.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/net_capital_gains/rate.yaml:1]:
```suggestion
description: Arizona subtracts this fraction of long term capital gains from adjusted gross income.
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    unit = USD
+    documentation = "https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i.pdf#page=31"
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.net_capital_gains
+
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/net_capital_gains/az_long_term_capital_gains_subtraction.py:17]:
```suggestion
    # Arizona adds back capital losses, so this is uncapped.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: could you also make a subtractions.yaml file and include this subtraction as the first one. We will fill it as other subtractions are merged.

There are examples in other state model
Let me know if that makes sense

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**